rerun_py/src/remote.rs

```diff
@@ -65,6 +70,120 @@ impl PyConnection {
                 .collect())
         })
     }
+
+    /// Register a recording along with some metadata
+    #[pyo3(signature = (
+        storage_url,
+        metadata = None
+    ))]
+    fn register(
+        &mut self,
+        storage_url: &str,
+        metadata: Option<&Bound<'_, PyDict>>,
+    ) -> PyResult<String> {
+        self.runtime.block_on(async {
+            let storage_url = url::Url::parse(storage_url)
+                .map_err(|err| PyRuntimeError::new_err(err.to_string()))?;
+
+            let _obj = object_store::ObjectStoreScheme::parse(&storage_url)
+                .map_err(|err| PyRuntimeError::new_err(err.to_string()))?;
+
+            let metadata = metadata
+                .map(|metadata| {
+                    let (schema, data): (
+                        Vec<arrow2::datatypes::Field>,
+                        Vec<Box<dyn arrow2::array::Array>>,
+                    ) = metadata
+                        .iter()
+                        .map(|(key, value)| {
+                            let key = key.to_string();
+                            let value = value.extract::<MetadataLike>()?;
+                            let value_array = value.to_arrow2()?;
+                            let field = arrow2::datatypes::Field::new(
+                                key,
+                                value_array.data_type().clone(),
+                                true,
+                            );
+                            Ok((field, value_array))
+                        })
+                        .collect::<PyResult<Vec<_>>>()?
+                        .into_iter()
+                        .unzip();
+
+                    let schema = arrow2::datatypes::Schema::from(schema);
+
+                    let data = arrow2::chunk::Chunk::new(data);
+
+                    let metadata_tc = TransportChunk {
+                        schema: schema.clone(),
+                        data,
+                    };
+
+                    RecordingMetadata::try_from(EncoderVersion::V0, &metadata_tc)
+                        .map_err(|err| PyRuntimeError::new_err(err.to_string()))
+                })
+                .transpose()?;
+
+            let request = RegisterRecordingRequest {
+                // TODO(jleibs): Description should really just be in the metadata
```

Comment:
you're suggesting that description is basically and optional metadata field, right? if so, that indeed makes sense to me.